spec/unit/indirector/file_bucket_file/file_spec.rb

```diff
@@ -92,32 +92,37 @@ def save_bucket_file(contents, path = "/who_cares")
 
       describe "when supplying a path" do
         with_digest_algorithms do
-          it "should store the path if not already stored" do
-            checksum = save_bucket_file(plaintext, "/foo/bar")
-
-            dir_path = "#{Puppet[:bucketdir]}/#{bucket_dir}"
-            contents_file = "#{dir_path}/contents"
-            paths_file = "#{dir_path}/paths"
-            expect(Puppet::FileSystem.binread(contents_file)).to eq(plaintext)
-            expect(Puppet::FileSystem.read(paths_file)).to eq("foo/bar\n")
-          end
-
-          it "should leave the paths file alone if the path is already stored" do
-            checksum = save_bucket_file(plaintext, "/foo/bar")
-            checksum = save_bucket_file(plaintext, "/foo/bar")
-            dir_path = "#{Puppet[:bucketdir]}/#{bucket_dir}"
-            expect(Puppet::FileSystem.binread("#{dir_path}/contents")).to eq(plaintext)
-            expect(File.read("#{dir_path}/paths")).to eq("foo/bar\n")
-          end
-
-          it "should store an additional path if the new path differs from those already stored" do
-            checksum = save_bucket_file(plaintext, "/foo/bar")
-
-            checksum = save_bucket_file(plaintext, "/foo/baz")
-
-            dir_path = "#{Puppet[:bucketdir]}/#{bucket_dir}"
-            expect(Puppet::FileSystem.binread("#{dir_path}/contents")).to eq(plaintext)
-            expect(File.read("#{dir_path}/paths")).to eq("foo/bar\nfoo/baz\n")
+          if !(Puppet::Util::Platform.windows? &&
+                ("#{digest_algorithm}" == "sha512" || "#{digest_algorithm}" == "sha384"))
```

Comment:
This would be much easier as an inline comment.

There's a `skip` test helper for this kind of thing. You could then do
```
skip('long SHAs don't work on Windows') if Puppet::Util::Platform.windows? && (digest_algorithm == 'sha512' || digest_algorithm == 'sha384')
```

Not sure why the logic itself wouldn't work, unless your failure really happens somewhere else.

Reply:
I narrowed down the issue to the use of "#{digest_algorithm}". I have not tried putting in a skip there since the if condition is needed (which seems to be the problem). The weird thing is there are multiple other instances that use that. I will experiment further and see.  thanks,